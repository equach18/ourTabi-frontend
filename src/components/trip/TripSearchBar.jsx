import { useState } from "react";
import { useNavigate } from "react-router-dom";

function TripSearchBar() {
  const [searchTerm, setSearchTerm] = useState("");
  const navigate = useNavigate();

  function handleSubmit(evt) {
    evt.preventDefault();
    if (searchTerm.trim()) {
      navigate(`/search?term=${encodeURIComponent(searchTerm.trim())}`);
    }
    setSearchTerm("");
  }

  return (
    <form onSubmit={handleSubmit} className="flex items-center space-x-2">
      <input
        type="text"
        placeholder="Search by location or title"
        value={searchTerm}
        onChange={(e) => setSearchTerm(e.target.value)}
        className="w-64 bg-white px-2 py-1 border rounded-md text-sm"
      />
      <button
        type="submit"
        className="bg-emerald-600 text-white px-3 py-1 rounded-md text-sm hover:bg-emerald-700"
      >
        Go
      </button>
    </form>
  );
}

export default TripSearchBar;
