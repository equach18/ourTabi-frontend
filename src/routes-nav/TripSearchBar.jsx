import { useState } from "react";
import { useNavigate } from "react-router-dom";

function TripSearchBar() {
  const [searchTerm, setSearchTerm] = useState("");
  const navigate = useNavigate();

  function handleSubmit(evt) {
    evt.preventDefault();
    if (searchTerm.trim()) {
      navigate(`/search?term=${encodeURIComponent(searchTerm.trim())}`);
    }
  }

  return (
    <form onSubmit={handleSubmit} className="flex items-center space-x-2">
      <input
        type="text"
        placeholder="Search trips..."
        value={searchTerm}
        onChange={(e) => setSearchTerm(e.target.value)}
        className="px-2 py-1 border rounded-md text-sm"
      />
      <button
        type="submit"
        className="bg-blue-500 text-white px-3 py-1 rounded-md text-sm hover:bg-blue-600"
      >
        Go
      </button>
    </form>
  );
}

export default TripSearchBar;
