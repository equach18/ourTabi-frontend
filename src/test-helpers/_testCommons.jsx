import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import UserContext from "../context/UserContext";
import TripContext from "../context/TripContext";

export const testUser1 = {
  id: 1,
  firstName: "Test1",
  lastName: "User1",
  username: "testUser1",
  profilePic: "",
};
export const testUser2 = {
  id: 2,
  firstName: "Test2",
  lastName: "User2",
  username: "testUser2",
  profilePic: "testUser2.png",
};
export const testUser3 = {
  id: 3,
  firstName: "Test2",
  lastName: "User2",
  username: "testUser3",
  profilePic: "testUser3.png",
};

export const defaultMembers = [
  {
    userId: testUser1.id,
    username: testUser1.username,
    profilePic: testUser1.profilePic,
  },
  {
    userId: testUser2.id,
    username: testUser2.username,
    profilePic: testUser2.profilePic,
  },
];

export const testActivity = {
  id: 1,
  name: "Kayaking",
  tripId: 53,
  category: "Other",
  location: "Chicago",
  description: "So much fun",
  scheduledTime: "2025-07-10T14:30:00Z",
};

export const testComment = {
  id: 1,
  text: "Test comment",
  createdAt: "2025-07-11T14:30:00Z",
  userId: 1,
};

export const testTrip = {
  id: 42,
  title: "Test trip",
  destination: "Chicago, IL",
  startDate: "2025-07-10",
  endDate: "2025-07-12",
  isPrivate: true,
};

export const testFriendsData = {
  friends: [testUser1 ],
  incomingRequests: [testUser2],
  sentRequests: [testUser3],
};

// Default mock context values
const defaultUserContext = { currentUser: testUser1 };
const defaultTripContext = {
  voteOnActivity: vi.fn(),
  votes: {},
  deleteComment: vi.fn(),
  addMember: vi.fn(),
  removeMember: vi.fn(),
  members: defaultMembers,
};

// create mock trip context
export function createMockTripContext(overrides = {}) {
  return {
    voteOnActivity: vi.fn(),
    votes: {},
    deleteComment: vi.fn(),
    members: [],
    addMember: vi.fn(),
    removeMember: vi.fn(),
    ...overrides,
  };
}

/**
 * Renders a component inside MemoryRouter
 */
export function renderWithRouter(ui) {
  return render(<MemoryRouter>{ui}</MemoryRouter>);
}

/**
 * Renders a component inside MemoryRouter + UserContext + TripContext
 */
export function renderWithAllProviders(
  ui,
  {
    userContext = defaultUserContext,
    tripContext = defaultTripContext,
    route = "/",
  } = {}
) {
  return render(
    <MemoryRouter initialEntries={[route]}>
      <UserContext.Provider value={userContext}>
        <TripContext.Provider value={tripContext}>{ui}</TripContext.Provider>
      </UserContext.Provider>
    </MemoryRouter>
  );
}

/**
 * Renders a component with just UserContext
 */
export function renderWithUserProvider(ui, { userContext } = {}) {
  return render(
    <UserContext.Provider value={userContext}>{ui}</UserContext.Provider>
  );
}
