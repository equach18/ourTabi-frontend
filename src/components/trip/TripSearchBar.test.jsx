import { fireEvent, screen } from "@testing-library/react";
import { Routes, Route } from "react-router-dom";
import TripSearchBar from "./TripSearchBar";
import { renderWithRouter } from "../../test-helpers/_testCommons";
// mock useNavigate
const mockNavigate = vi.fn();

vi.mock("react-router-dom", async () => {
  const actual = await vi.importActual("react-router-dom");
  return {
    ...actual,
    useNavigate: () => mockNavigate,
  };
});

function renderSearchBar() {
  return renderWithRouter(
    <Routes>
      <Route path="*" element={<TripSearchBar />} />
    </Routes>
  );
}

describe("TripSearchBar", () => {
  beforeEach(() => {
    mockNavigate.mockClear();
  });

  it("matches snapshot", () => {
    const { asFragment } = renderSearchBar();
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders input and button", () => {
    renderSearchBar();

    expect(
      screen.getByPlaceholderText(/search by location or title/i)
    ).toBeInTheDocument();
    expect(screen.getByRole("button", { name: /go/i })).toBeInTheDocument();
  });

  it("navigates to correct URL on valid input", () => {
    renderSearchBar();

    const input = screen.getByPlaceholderText(/search by location or title/i);
    fireEvent.change(input, { target: { value: "chicago" } });

    fireEvent.click(screen.getByRole("button", { name: /go/i }));

    expect(mockNavigate).toHaveBeenCalledWith("/search?term=chicago");
  });

  it("does not navigate on empty input", () => {
    renderSearchBar();

    fireEvent.click(screen.getByRole("button", { name: /go/i }));

    expect(mockNavigate).not.toHaveBeenCalled();
  });

  it("clears input after submit", () => {
    renderSearchBar();

    const input = screen.getByPlaceholderText(/search by location or title/i);
    fireEvent.change(input, { target: { value: "Japan" } });

    fireEvent.click(screen.getByRole("button", { name: /go/i }));

    expect(input.value).toBe(""); 
  });
});
