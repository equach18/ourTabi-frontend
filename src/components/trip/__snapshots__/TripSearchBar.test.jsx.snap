// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TripSearchBar > matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="flex items-center space-x-2"
  >
    <input
      class="w-64 bg-white px-2 py-1 border rounded-md text-sm"
      placeholder="Search by location or title"
      type="text"
      value=""
    />
    <button
      class="bg-emerald-600 text-white px-3 py-1 rounded-md text-sm hover:bg-emerald-700"
      type="submit"
    >
      Go
    </button>
  </form>
</DocumentFragment>
`;
